middle_end/flambda2/simplify/env/downwards_env.ml

```diff
@@ -543,8 +544,12 @@ let enter_inlined_apply ~called_code ~apply ~was_inline_always t =
         in
         Inlining_state.increment_depth t.inlining_state ~by)
   in
+  let inlined_debuginfo =
+    Inlined_debuginfo.create ~called_code_id:(Code.code_id called_code)
+      ~apply_dbg:(Apply.dbg apply)
+  in
```

Comment:
I'm not sure what happens when inlining from classic mode and simplify interleave, and I suspect we could be forgetting/discarding some debuginfos, particularly here since we simply overwrite the current `inlined_debuginfo`: imagine the current value of `t.inlined_debuginfo` comes from `Enter_inlined_apply` primitive inserted during classic mode, what ensures that we do not forget the inlined debuginfo from those when we overwrite it in this function ?

Maybe just a comment/CR about this should be enough for now.

Reply:
As discussed this is likely a pre-existing problem; have added a CR.